### ISSUE TITLE
Fails to build with MSVC

### ISSUE THREAD

**Issue by KarolS** [pyresidfp/pyresidfp#1]
Doing pip install -U pyresidfp yields the following:

```
  FAILED: CMakeFiles/_pyresidfp.dir/src/residfp/resample/SincResampler.cpp.obj
  "D:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\bin\Hostx86\x64\cl.exe"  /nologo /TP -DHAVE_CONFIG_H -DPROJECT_VERSION -D_pyresidfp_EXPORTS -IC:\Users\Karol\AppData\Local\Temp\pip-install-fpar1ybc\pyresidfp\_skbuild\win-amd64-3.9\cmake-build -IC:\Users\Karol\AppData\Local\Temp\pip-install-fpar1ybc\pyresidfp -IC:\Users\Karol\AppData\Local\Temp\pip-install-fpar1ybc\pyresidfp\src\residfp -IC:\Users\Karol\AppData\Local\Temp\pip-install-fpar1ybc\pyresidfp\src\residfp\resample -I"d:\program files\python39\Include" -IC:\Users\Karol\AppData\Local\Temp\pip-build-env-dw84bquk\overlay\Lib\site-packages\pybind11\include /DWIN32 /D_WINDOWS /GR /EHsc /O2 /Ob2 /DNDEBUG -MD /bigobj /MP /GL /showIncludes /FoCMakeFiles\_pyresidfp.dir\src\residfp\resample\SincResampler.cpp.obj /FdCMakeFiles\_pyresidfp.dir\ /FS -c C:\Users\Karol\AppData\Local\Temp\pip-install-fpar1ybc\pyresidfp\src\residfp\resample\SincResampler.cpp
  C:\Users\Karol\AppData\Local\Temp\pip-install-fpar1ybc\pyresidfp\_skbuild\win-amd64-3.9\cmake-build\siddefs-fp.h(34): fatal error C1017: invalid integer constant expression
```

**Comment by packet23** [comment]
Hey @KarolS , thanks for reporting. The bug will be difficult for me to reproduce and fix (don't have access to a windows machine setup for python development). I'll see what I can do, but would greatly appreciate a pull request.

**Comment by packet23** [comment]
I think I fixed the build. At least it builds on Github. @KarolS would you mind checking if PR #2 builds on your machine as well?

**Comment by KarolS** [comment]
Hey @packet23 
Sorry for such a bad issue report, I just posted it reflexively without much thinking. I didn't even specify which version of pyresidfp I was trying to use (it was 0.4). I hacked around that issue, but I didn't get it to work, but that doesn't matter at all now.

Currently, I'm getting `Both setup.py and pyproject.toml have setuptools-git-versioning config. Please remove one of them`

So I set `version_config=False, setup_requires=[],` in setup.py and removed `"setuptools-git-versioning",` in pyproject.toml as a temporary workaround.

And it works. It just compiles and works, and desidulate (which is what I wanted to use) also works.

Thanks!

**Comment by packet23** [comment]
Hey @KarolS , I believe this additional issue has been addressed in release 0.5.1. Also, binary (wheel) packages are available from pypi as well, for instance here for Python 3.10: https://files.pythonhosted.org/packages/28/c6/d14200a3c489f6385e4db196e83d0b4916494839f31b88e69e038ef2bae9/pyresidfp-0.5.1-cp310-cp310-win_amd64.whl .
